### PR TITLE
schema/schemaspec: support string and bool lists

### PR DESCRIPTION
**Pull request by rotemtam**
More work towards supporting primitive types in HCL lists. 

Next PR will include support for numeric types.

### CODE REVIEW

File: schema/schemaspec/extension.go [hunk 1]
```diff
@@ -320,17 +341,32 @@ func scanAttr(key string, r *Resource, field reflect.Value) error {
...
+			item := field.Index(i).Interface().(string)
+			lst.V = append(lst.V, &LiteralValue{V: strconv.Quote(item)})
+		}
+	case reflect.Bool:
+		for i := 0; i < field.Len(); i++ {
+			item := field.Index(i).Interface().(bool)
+			lst.V = append(lst.V, &LiteralValue{V: strconv.FormatBool(item)})
+		}
+	case reflect.Ptr:
+		if typ.Kind() != reflect.Slice || typ.Elem() != reflect.TypeOf(&Ref{}) {
```
Comment (yonidavidson) [inline schema/schemaspec/extension.go:361]: not a fan of this kind of logic, I prefer to break it but make it more readble.

Comment (yonidavidson) [reply schema/schemaspec/extension.go:361]: even better, error log by case.

File: sql/mysql/sqlspec_test.go [hunk 2]
```diff
@@ -454,6 +454,7 @@ func TestMarshalSpecColumnType(t *testing.T) {
 			}
 			s.Tables[0].Schema = &s
 			ddl, err := MarshalSpec(&s, schemahcl.Marshal)
+			fmt.Println(string(ddl))
```
Comment (yonidavidson) [inline sql/mysql/sqlspec_test.go:457]: why print?

File: schema/schemaspec/extension.go (hunk 1 above)
Comment (rotemtam) [reply schema/schemaspec/extension.go:361]: good point the assertion that its a slice is not needed at all, dropping it

File: sql/mysql/sqlspec_test.go (hunk 2 above)
Comment (rotemtam) [reply sql/mysql/sqlspec_test.go:457]: correct!~

**Review by a8m — APPROVED** [review]

**Pull request merged**